File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*;

class Solution {
    public int[] solution(int n, long left, long right) {
        int[] answer;        
        
        // System.out.println(Arrays.deepToString(arr));
        answer = new int[(int)(right - left + 1)];
        for(long i = left; i <= right; i++){
            int row = (int)(i / n) + 1;
            int col = (int)(i % n) + 1;
            answer[(int)(i -left)] = Math.max(row, col);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
def solution(n, left, right):
    # 1 2 3 4
    # 2 2 3 4
    # 3 3 3 4
    # 4 4 4 4
    
    def calc_number_on_square(n, number):
        row = number // n + 1
        col = number % n + 1
        
        if col <= row:
            return row
        else:
            return col
                
    
    answer = [calc_number_on_square(n, num) for num in range(left, right+1)]
    return answer